src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -407,6 +408,63 @@ uct_cuda_ipc_mem_dereg(uct_md_h md, const uct_md_mem_dereg_params_t *params)
     return UCS_OK;
 }
 
+static int
+uct_cuda_ipc_md_check_fabric_info(uct_cuda_ipc_md_t *md,
+                                  ucs_ternary_auto_value_t mnnvl_enable)
+{
+    int mnnvl_supported = 0;
+#if HAVE_NVML_FABRIC_INFO
+    nvmlGpuFabricInfoV_t fabric_info;
+    nvmlDevice_t device;
+    ucs_status_t status;
+    char buf[64];
+
+    if (!uct_cuda_base_is_coherent() || (mnnvl_enable == UCS_NO)) {
```

Comment:
what if we didnt check uct_cuda_base_is_coherent here?

Reply:
Better to return early from this function and not go through nvml calls if platform is not coherent. MNNVL is only supported on coherent platforms.

Reply:
this is what happens now

Reply:
Spoke to Yossi offline. We can just do the nvml checks as these checks are performed once per MD and is off the critical path. It maybe worth profiling this function overall just to keep tabs on nvml call overheads as they are generally on the higher side.

Reply:
fixed